src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -104,6 +106,19 @@ export default function getAdaptationSwitchStrategy(
     return { type: "needs-reload", value: undefined };
   }
 
+  if (adaptation.type === "audio" &&
+      // We have been explicitly asked to reload
+      audioTrackSwitchingMode === "direct" &&
+      // We're playing the current Period
+      isTimeInRange({ start, end }, currentTime) &&
+      // There is data for the current position
+      playbackInfo.readyState > 1 &&
+      // We're not playing the current wanted audio Adaptation yet
+      !isTimeInRanges(adaptationInBuffer, currentTime))
+  {
```

Comment:
Nice.

I propose that we can also ensure that the current codec is compatible.
Something like `(playbackInfo.readyState > 1 || areCodecsCompatible(segmentBuffer.codec, codec))` as we want to reload if either there is playing data for the current Period (the readyState condition part) but also if there is none but the current audio SourceBuffer was created for another codec.

As is, we might not reload when switching quickly (before any segment has been pushed) between two Adaptations with different codecs. This could be problematic.
What do you think?

We might also replicate the same condition in the video condition.

Reply:
Ha yes, I see what u mean.

Yea, then, a utils function that checks that the current adaptation and the next adaptation codecs are the same would make sense for me.

However, I can only get codecs on the representations, but I guess, we said that it never could happen a day where multiple different codecs are present in the same adaptation right?

But what if someday, we encounter that?

Reply:
I've pushed a condition where I'm checking that if at least one codec is different in all the playable representations.

However, I'm not sure about the utils function `areCodecsCompatible` because all she does is only equality.  
But its clearer for readability, as the name of the function, helps us understand our will

Reply:
Yes that's a good question and it's definitely something problematic that should be fixed at some point.
There are many ways we could fix that and I'm not sure of the best way to do it.

I think the whole logic behind codec switch could be the object of another PR.
But this would mean that right now, I'm not sure of what we should do here.

Either we add that secondary condition with a `TODO` on top to explain that we should find a better solution.
Either we do nothing and wait for that future PR.

Right now, because there are real cases where not reloading could be problematic (like switching to a DolbyDigital+ track directly on the PS5), I think we can add it temporarily.

But you're right that this is not a good solution.

Reply:
>!adaptation.getPlayableRepresentations()
        .some(rep => areCodecsCompatible(rep.getMimeTypeString(), segmentBuffer.codec ?? ""))

I like your solution better than what I would have proposed (just to check the first Representation)!